modules/ROOT/pages/dataweave-formats-flatfile.adoc

```diff
@@ -29,8 +29,116 @@ payload
-include::partial$dataweave-formats-flatfile-config.adoc[]
+
+[[properties]]
+== Configuration Properties
+
+DataWeave supports the following configuration properties for this format.
+
+
+[[reader_properties]]
+=== Reader Properties
+
+This format accepts properties that provide instructions for reading input data.
+
+
+[cols="1,1,1,3a", options="header"]
+|===
+|Parameter |Type |Default|Description
+|`allowLenientWithBinaryNotEndElement`  |`Boolean`|`false`|When the schema contains elements of type Binary or Packed, the `lenient` option does not allow short records, regardless of the last element's
+format type. When you set this property to `true`, the validation applies only to cases in which the record ends with type Binary or Packed.
+
+Valid values are `true` or `false`.
+|`enforceRequires`  |`Boolean`|`false`|Produces an error when set to `true` if a required value is missing.
+
+Valid values are `true` or `false`.
+|`missingValues`  |`String`|`null`|Fill character used to represent missing values. To activate a non-default setting, set the `useMissCharAsDefaultForFill` property to `true`, and use one of the following values to `missingValues`:
+
+* `none` (for the reader) or `NONE` (for the writer): Treats all data as values.
+* `spaces` (for the reader) or `SPACES` (for the writer): Interprets a field consisting of only spaces as a missing value. Default for flat file and fixed-width formats.
+* `zeroes` (for the reader) or `ZEROES` (for the writer): Interprets numeric fields consisting of only `0` characters _and_ character fields consisting of only spaces as missing values.
+* `nulls` (for the reader) or `NULLS` (for the writer): Interprets a field consisting only of `0` bytes as a missing value. Default for COBOL copybook schema.
+
+Valid values are `none` or `spaces` or `zeroes` or `nulls`.
+|`notTruncateDependingOnSubjectNotPresent`  |`Boolean`|`false`|Fills the entire group when the DEPENDING ON subject is not present.
+
+Valid values are `true` or `false`.
+|`recordParsing`  |`String`|`'strict'`|Specifies the expected type of separation between lines or records:
+
+* `strict`: Line break is expected at exact end of each record. `strict` is the default.
+* `lenient`: Line break is used, but records can be shorter or longer than the schema specifies.
+* `noTerminator`: Records follow one another with no separation.
+* `singleRecord`: The entire input is a single record.
+
+Note that schemas with type `Binary` or `Packed` don't allow for line break detection, so setting `recordParsing` to `lenient` only allows long records to be handled, not short ones. These schemas also currently only work with certain single-byte character encodings (so not with UTF-8 or any multibyte format).
+
+Valid values are `strict` or `lenient` or `noTerminator` or `singleRecord`.
+|`retainEmptyStringFieldsOnParsing`  |`Boolean`|`false`|Allow parsing behavior to keep missing string value fields with a default value in the output map
```

Comment:
`retainEmptyStringFieldsOnParsing` not in 2.4. Is this new? Check description and which of the 3 formats it applies to.

Reply:
Introduced in https://docs.mulesoft.com/release-notes/mule-runtime/mule-4.3.0-release-notes#january-2023

W-12036336

There's a separate PR with this for the v2.4 (and v2.3 doc): https://github.com/mulesoft/docs-dataweave/pull/218

Reply:
TODO: add to include

Reply:
handling the rest of this via #218